src/client.rs

```diff
@@ -96,6 +96,15 @@ impl Client {
         Ok(Self::new(connector, None, endpoint))
     }
 
+    /// Create a connection to APNs using the raw PEM-formatted certificate and
+    /// key, extracted from the provider client certificate you obtain from your
+    /// [Apple developer account](https://developer.apple.com/account/).
+    pub fn certificate_parts(cert: &[u8], key: &[u8], endpoint: Endpoint) -> Result<Client, Error> {
```

Comment:
Could the `cert` and `key` be `impl Read` instead, allowing one to load them directly from files? `&[u8]` should also implement `Read`.

Reply:
[`AlpnConnector::with_client_cert`](https://docs.rs/hyper-alpn/0.2.1/hyper_alpn/struct.AlpnConnector.html#method.with_client_cert) takes in `&[u8]` so we don't benefit from using `impl Read` here.